cmd/common/utils/helpers.go

```diff
@@ -0,0 +1,191 @@
+		return "", fmt.Errorf("no scheme allowed")
+	}
+
+	if len(image) == 0 {
+		return "", fmt.Errorf("empty image")
+	}
+	index := strings.LastIndexByte(image, ':')
+	// image without a tag
+	if index == -1 {
+		return "latest", nil
+	}
+	tag := image[index+1:]
+	if !isValidTag(tag) {
+		return "", fmt.Errorf("%q is not a valid tag", tag)
+	}
+	return tag, nil
+}
+
+func GetRepositoryFromImage(image string) (string, error) {
+	// For now just assume its something like ghcr.io:443/inspektor-gadget/ig:latest
+	// So we need everything before the last colon, which includes also the port
+	url, err := url.Parse(image)
+	if err != nil {
+		return "", fmt.Errorf("parse url %q: %w", image, err)
+	}
+	if url.Host != "" && url.Scheme != "" {
+		return "", fmt.Errorf("no scheme allowed")
+	}
+
+	if len(image) == 0 {
+		return "", fmt.Errorf("empty image")
+	}
+	index := strings.LastIndexByte(image, ':')
+	// image without a tag
+	if index == -1 {
+		return image, nil
+	}
+	tag := image[index+1:]
+	if !isValidTag(tag) {
+		return "", fmt.Errorf("%q is not a valid tag", tag)
+	}
+	return image[:index], nil
+}
+
+func NormalizeImage(image string) (string, error) {
+	tag, err := GetTagFromImage(image)
+	if err != nil {
+		return "", fmt.Errorf("get tag from image %q: %w", image, err)
+	}
+	repository, err := GetRepositoryFromImage(image)
+	if err != nil {
+		return "", fmt.Errorf("get repository from image %q: %w", image, err)
+	}
+	return fmt.Sprintf("%s:%s", repository, tag), nil
+}
+
+func GetHostString(repository string) (string, error) {
+	// We need everything before the first slash, which includes also the port, but remove the scheme
+	url, err := url.Parse(repository)
+	if err != nil {
+		return "", fmt.Errorf("parse url %q: %w", repository, err)
+	}
+	if url.Host != "" && url.Scheme != "" {
+		return "", fmt.Errorf("no scheme allowed")
+	}
+
+	if len(repository) == 0 {
+		return "", fmt.Errorf("empty repo url")
+	}
+	// mydomain.io could be a repository. So tolerate a string without "/"
+	return strings.Split(repository, "/")[0], nil
+}
+
+func SetupAuthVariablesAndFlags(cmd *cobra.Command, authOptions *AuthOptions) {
+	// Flag inspired by https://github.com/containers/common/blob/cac40138f7e3c2b29ca32e64348535516bf6aa51/pkg/auth/cli.go#L48
+	cmd.Flags().StringVar(&authOptions.AuthFile, "authfile", DefaultAuthFile,
+		"path of the authentication file. This overrides the REGISTRY_AUTH_FILE environment variable")
+	viper.BindPFlag("registry.auth_file", cmd.Flags().Lookup("authfile"))
+	viper.BindEnv("registry.auth_file", "REGISTRY_AUTH_FILE")
+}
+
+func CreateAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.Client, error) {
+	logrus.Debugf("Using auth file %q", authOptions.AuthFile)
+
+	var cfg *configfile.ConfigFile
+	var err error
+
+	if authOptions.AuthFile == "" {
+		cfg, err = config.Load(authOptions.AuthFile)
+		if err != nil {
+			return nil, fmt.Errorf("load auth config: %w", err)
+		}
+	} else {
```

Comment:
I see you removed the `use-docker` option, what's the plan there? By using something like 

```diff
diff --git cmd/common/utils/helpers.go cmd/common/utils/helpers.go
index ab7071fd..93394d79 100644
--- cmd/common/utils/helpers.go
+++ cmd/common/utils/helpers.go
@@ -153,9 +153,14 @@ func CreateAuthClient(repository string, authOptions *AuthOptions) (*oras_auth.C
 
        var cfg *configfile.ConfigFile
        var err error
+       var found bool
 
-       if authOptions.AuthFile == "" {
-               cfg, err = config.Load(authOptions.AuthFile)
+       if _, err := os.Stat(authOptions.AuthFile); err == nil {
+               found = true
+       }
+
+       if !found {
+               cfg, err = config.Load("")
                if err != nil {
                        return nil, fmt.Errorf("load auth config: %w", err)
                }
```

we can make it default to use docker configuration if the ig one is not available.

Reply:
I misread a review comment from earlier :D

Reply:
Looks good to me. I won't worry too much about the $HOME issue, probably that's gone be changed when https://github.com/inspektor-gadget/inspektor-gadget/issues/1681 gets in.